src/main/kotlin/no/ssb/metadata/vardef/security/VardefLabidTokenValidator.kt

```diff
@@ -0,0 +1,169 @@
+
+    private fun getActiveGroup(token: JWT): String? = token.jwtClaimsSet.getClaim(activeGroupClaim) as? String
+
+    private fun usernameFromToken(token: JWT): String? =
+        (token.jwtClaimsSet.getClaim(usernameClaim) as? String)
+            ?.let { "$it@ssb.no" } // TODO: make constant or something so its not hardoceded here
+
+    /**
+     * @return `true` if the principal has specified a group which is not present in their token.
+     */
+    private fun activeGroupSpoofed(
+        activeGroup: String,
+        token: JWT,
+    ): Boolean = activeGroup !in getDaplaGroups(token)
+
+    /**
+     * @return `true` if the principal can be assigned the [VARIABLE_OWNER] role.
+     */
+    private fun isVariableOwner(claimsSet: JWTClaimsSet): Boolean =
+        claimsSet.getStringListClaim(Claims.AUDIENCE).any {
+            it in allowedAudiences
+        }
+
+    /**
+     * @return `true` if the principal can be assigned the [VARIABLE_CREATOR] role.
+     */
+    private fun isVariableCreator(
+        activeGroup: String,
+        claimsSet: JWTClaimsSet,
+    ): Boolean = isVariableOwner(claimsSet) && DaplaTeamService.isDevelopers(activeGroup)
+
+    /**
+     * @return `true` the token and request contain the fields necessary to assign roles.
+     */
+    private fun tokenAndRequestContainExpectedFields(
+        token: JWT,
+        request: R,
+    ): Boolean =
+        activeGroupClaim in token.jwtClaimsSet.claims &&
+            daplaGroupsClaim in token.jwtClaimsSet.claims &&
+            getDaplaGroups(token).isNotEmpty()
+
+    /**
+     * Assign roles
+     *
+     * The roles are assigned based on claims in the token and the `active_group` query parameter.
+     *
+     * The token is expected to have claims added by `oidc-dapla-userinfo-mapper` with the
+     * `nested_teams` config set to `true`. Ref <https://github.com/statisticsnorway/keycloak-iac/blob/4fb230adb60412e7a546612fec9e5b9903400825/pkl/GenericClient.pkl#L160>
+     *
+     * By default, authenticated users receive the [VARIABLE_CONSUMER] role. If they fulfill the
+     * requirements then they receive other roles in addition.
+     *
+     * @param token the parsed JWT token
+     * @param request the [HttpRequest]
+     * @return the set of applicable roles.
+     * @throws InvalidActiveGroupException
+     */
+    private fun assignRoles(
+        token: JWT,
+        request: R,
+    ): Set<String> {
+        // If we arrive here the principal is authenticated. Give all principals a default role.
+
+        val roles = mutableSetOf(VARIABLE_CONSUMER)
+        val claimsSet = token.jwtClaimsSet
+        val activeGroup = getActiveGroup(token) ?: return roles.toSet()
+        if (tokenAndRequestContainExpectedFields(token, request)) {
+            if (activeGroupSpoofed(activeGroup, token)) {
+                throw InvalidActiveGroupException("The specified active_group is not present in the token")
+            }
```

Comment:
```suggestion
```

This shouldn't be necessary for LabID since the active group comes from a claim in a signed token.